src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -195,7 +199,8 @@ UInt64 MergeTreeDataMergerMutator::getMaxSourcePartSizeForMutation()
     UInt64 disk_space = data.getStoragePolicy()->getMaxUnreservedFreeSpace();
 
     /// Allow mutations only if there are enough threads, leave free threads for merges else
-    if (background_pool_size - busy_threads_in_pool >= data_settings->number_of_free_entries_in_pool_to_execute_mutation)
+    if (busy_threads_in_pool <= 1
```

Comment:
What will be in case if `background_pool_size` is too lower than `number_of_free_entries_in_pool_to_execute_mutation`? I think that we will execute mutation with a few threads in pool (and we will use the **whole** pool), so it will be processed significantly slower that usually and we won't be able to execute any merges or other mutations for a period of time. If there are many inserts, it will  lead to `too many parts` error maybe.

Reply:
I'll merge this because it makes things better...

Reply:
@nikitamikhaylov #13153